chapter-05/final/money/decimal.go

```diff
@@ -0,0 +1,128 @@
+package money
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// number is a structure that can hold a number with a fixed precision.
+// example: 1.52 = 1 + 52 * 10^(-2) will be stored as {1, 52, 2}
+type number struct {
+	// the integer part of the number
+	integerPart int
+	// the decimal part of the number
+	decimalPart int
+	// precision of the decimal part, as the exponent of a power of 10
+	precision int
```

Comment:
It avoids casting `byte` to `int` when using `math.Pow10`. We find it more readable to use `math.Pow10` than the map:
```
var pow10 = []int64{
	0: 1,
	1: 10,
	2: 100,
	3: 1000,
	4: 10000,
	5: 100000,
	6: 1000000,
	7: 10000000,
	8: 100000000,
	9: 1000000000,
}
```

Reply:
The reason I would recommend `byte` and using smaller values in such scenarios is because such types can end up being used in larger computation. But, I don't think it's a big difference.